mmeval/metrics/bleu.py

```diff
@@ -0,0 +1,193 @@
+            references_len_diff = [
+                abs(len(prediction) - length) for length in references_len_list
+            ]
+            # In the multi sentence reference, the one whose length is closest
+            # to the predicted sentence is selected to record the length.
+            min_index = references_len_diff.index(min(references_len_diff))
+            references_len += references_len_list[min_index]
+
+            pred_counter: Counter = get_n_gram(prediction, self.n_gram)
+            reference_counter: Counter = Counter()
+            for reference in references:
+                # Take intersection for the n_gram of references.
+                reference_counter |= get_n_gram(reference, self.n_gram)
+            # Union the n_gram of prediction and references.
+            counter_clip = pred_counter & reference_counter
+
+            for counter in counter_clip:
+                precision_matches[len(counter) - 1] += counter_clip[counter]
+            for counter in pred_counter:
+                precision_total[len(counter) - 1] += pred_counter[counter]
+        result = (pred_len, references_len, precision_matches, precision_total)
+        self._results.append(result)
+
+    def compute_metric(
+            self, results: List[Tuple[int, int, np.ndarray,
+                                      np.ndarray]]) -> dict:
+        """Compute the bleu metric.
+
+        This method would be invoked in ``BaseMetric.compute`` after
+        distributed synchronization.
+
+        Args:
+            results (List[Tuple[int, int, np.ndarray, np.ndarray]]):
+                A list that consisting the tuple of correct numbers.
+                Tuple contains pred_len, references_len,
+                precision_matches, precision_total.
+                This list has already been synced across all ranks.
+
+        Returns:
+            Dict[str, float]: The computed bleu score.
+        """
+        pred_len = 0
+        references_len = 0
+        precision_matches = np.zeros(self.n_gram)
+        precision_total = np.zeros(self.n_gram)
+        for result in results:
+            pred_len += result[0]
+            references_len += result[1]
+            precision_matches += result[2]
+            precision_total += result[3]
+
+        if min(precision_matches) == 0.0:
+            return {'bleu': np.array(0.0)}
+        if self.smooth:
+            precision_score = np.add(precision_matches, np.ones(
+                self.n_gram)) / np.add(precision_total, np.ones(self.n_gram))
+            precision_score[0] = precision_matches[0] / precision_total[0]
+        else:
+            precision_score = precision_matches / precision_total
+
+        precision_score = np.array(
+            self.ngram_weights) * np.log(precision_score)
+        brevity_penalty = _get_brevity_penalty(pred_len, references_len)
+        bleu = brevity_penalty * np.exp(np.sum(precision_score))
+        result = {'bleu': round(float(bleu), 6)}
```

Comment:
We tend to return the raw results.
```suggestion
        result = {'bleu': float(bleu)}
```